index.d.ts

```diff
@@ -10,6 +10,9 @@ declare module 'fastify' {
     > {
     ip: string | number
   }
+  interface FastifyInstance {
+    rateLimit: (options?:RateLimitPluginOptions) => preHandlerAsyncHookHandler;
```

Comment:
These are the same options as the plugin. From reading all of them it mostly makes sense but please let me know if you would like these options to be different.
For example, the `global` boolean does  not seem necessary.

Reply:
Then, use `Omit<RateLimitPluginOptions, 'global'>`

Reply:
I split the TS definitions instead to only expose the used options on `rateLimit(opts)` Hope that's ok.